Modify CopyDir task to leverage symfony Filesystem copy.

malikkotob opened the pull request:
### Overview
This pull request:

- [ ] Fixes a bug
- [x] Adds a feature
- [ ] Breaks backwards compatibility
- [ ] Has tests that cover changes

### Summary
Currently, the `CopyDir` task does not allow a user to specify whether or not destination files should be overwritten in the event that they are newer than the source files (a la the `overwrite` option provided in [Phing's copy task](https://www.phing.info/docs/guide/trunk/CopyTask.html). This pull request adds in an optional `$overwrite` parameter to the `CopyDir` task that defaults to `true` to ensure backwards compatibility (could consider changing the default to `false` for a major release).

### Description
To achieve the above behavior, `CopyDir` was modified to use the existing symfony `fs` object's `copy` method instead of what it is currently using - php's built in `copy` method.

greg-1-anderson commented:
Thanks for this enhancement.

Could you set `$overwrite` with a setter method rather than a constructor parameter?  Backwards compatibility is important for the `CopyDir` class as well as for `taskCopyDir`; switching to a setter method will also take care of this concern.

malikkotob commented:
Thanks for the quick response @greg-1-anderson. Good point. I went ahead and refactored to use a setter method and removed the attribute from the constructor.

greg-1-anderson commented:
Perfect -- thanks.
